src/OrchardCore/OrchardCore/Shell/ShellDescriptorFeaturesManager.cs

```diff
@@ -206,4 +246,20 @@ private List<IFeatureInfo> GetFeaturesToDisable(IFeatureInfo featureInfo, IEnume
 
         return featuresToDisable;
     }
+
+    private async Task<IEnumerable<string>> GetDeploymentStepNamesAsync()
```

Comment:
BTW, I'm thinking of introducing a new service to allow other parties to participate in feature dependency, which is good for extensibility. This way, the system will look into many places without touching this class anymore

Can I have a try? Or do you have a better suggestion

Reply:
That's why I suggesting to introduce a service to eliminate the dependency

Reply:
Please explain why `IFeatureEventHandler` isn't working. What kind of implementation would need to be duplicated?

Reply:
Can't you just re-enable the feature from an event handler implementation?